lib/spack/docs/frequently_asked_questions.rst

```diff
@@ -0,0 +1,62 @@
+.. Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+   Spack Project Developers. See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+==========================
+Frequently Asked Questions
+==========================
+
+-------------------------------------------------------
+What determines what versions and variants Spack picks?
+-------------------------------------------------------
+
+This question comes up in a variety of forms:
+
+ 1. Why does Spack ignore the preferences for versions / variants
+    that I set in my ``packages.yaml`` file?
+ 2. Why does Spack not use the default variant value from the
+    ``package.py`` file?
+
+The short answer is that Spack picks the optimal versions, variant
+values, and dependencies, and the optimal choice does not always
```

Comment:
Perhaps elaborate on `optimal`, such as "...picks the optimal versions, variants, values, and dependencies based on a complex set of criteria.  Therefore, the optimal choice does not always..."

Reply:
i've reworded things more generally